### PR TITLE
Add support for showExtensions on Response objects

### PR DESCRIPTION
**Pull request by shelbys**
If you appreciate this contribution, please add a `hacktoberfest-accepted` Label to the PR, so it'll be tracked

### Description
Add same ability to show `x-` extensions for Response objects, just like already supported for Operations, Parameters, and Schemas

### Motivation and Context
We actively use `x-` extensions on Response objects to better detail individual cases, but they're currently not shown in `swagger-ui

### How Has This Been Tested?
I manually verified locally with changes to PetStore, but didn't find any automated tests to extend.  Let me know if I missed anything

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [X] Improvements (misc. changes to existing features)
- [X] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [X] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [X] My changes require a change to the project documentation.
- [X] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [X] My changes can and should be tested by unit and/or integration tests.
- [X] If yes to above: I have added tests to cover my changes.
- [X] If yes to above: I have taken care to cover edge cases in my tests.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by tim-lai** [comment]
@shelbys Thanks for the PR! Functionally looks good. However, it should come with new tests 1) where using `x-` extension does in fact render, and 2) where not using `x-` extension does not render the Response Extension component.

File: src/core/components/response-extension.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from "react"
+import PropTypes from "prop-types"
+
+export const ResponseExt = ({ xKey, xVal }) => {
```
Comment (tim-lai) [inline src/core/components/response-extension.jsx:4]: let's use a fully descriptive name, such as `ResponseExtension`

**Review by tim-lai — CHANGES_REQUESTED** [review]

**Comment by shelbys** [comment]
@tim-lai - I'm happy to add some tests, just didn't find any to model off.  Do you have a pointer to a test I can use as a starting point?

**Comment by shelbys** [comment]
@tim-lai - Renamed to ResponseExtension, and tests added

**Review by tim-lai — APPROVED** [review]

**Comment by tim-lai** [comment]
@shelbys Merged! Thanks for the contribution!

**Pull request merged**